__tests__/app.test.js

```diff
@@ -281,6 +281,71 @@ describe("Articles", () => {
       });
     });
   });
+
+  describe("Get articles by queries", () => {
+    it("GET 200: returns articles filtered by topic value, if no query respond with all articles", () => {
+      return request(app)
+        .get("/api/articles?topic=cats")
+        .expect(200)
+        .then(({ body }) => {
+          // only one in test db
+          expect(body.articles).toHaveLength(1);
+        });
+    });
+    it("GET 200: returns articles filtered by topic value", () => {
+      return request(app)
+        .get("/api/articles?topic=mitch")
+        .expect(200)
+        .then(({ body }) => {
+          expect(body.articles).toHaveLength(11);
+        });
+    });
+    // sorting by author, title, topic, comment count, votes, created_at
+    it("GET 200: returns articles sorted by valid sorting options: author ASC", () => {
+      return request(app)
+        .get("/api/articles?sort_by=author&order=asc")
+        .expect(200)
+        .then(({ body }) => {
+          const articles = body.articles;
+          expect(articles).toBeSortedBy("author", { ascending: true });
+        });
+    });
+    it("GET 200: returns articles sorted by valid sorting options: Mitch comment_count desc", () => {
+      return request(app)
+        .get("/api/articles?topic=mitch&sort_by=comment_count")
+        .expect(200)
+        .then(({ body }) => {
+          const articles = body.articles;
+          expect(articles).toBeSortedBy("comment_count", { coerce: true, descending: true });
+        });
+    });
+    // tests for errors sort_by and order
+    it("GET 400: if invalid sort_by ", () => {
+      return request(app)
+        .get("/api/articles?sort_by=author_age")
+        .expect(400)
+        .then(({ body }) => {
+          expect(body.message).toBe('Bad request')
+        });
+    });
+    it("GET 400: if invalid order ", () => {
+      return request(app)
+        .get("/api/articles?sort_by=author&order=blackhole")
+        .expect(400)
+        .then(({ body }) => {
+          expect(body.message).toBe("Bad request");
+        });
+    });
+    it("GET 404: if article do not exist", () => {
+      return request(app)
+        .get("/api/articles?topic=monkeys")
+        .expect(404)
+        .then(({ body }) => {
+          console.log(body)
+          expect(body.message).toBe("Article not found");
+        });
+    });
+  });
```

Comment:
This is stunning work really thorough testing, and correct status codes throughout! You've perhaps missed one test, 200 Filtered by valid topic but there are no articles (returns empty array), I think the "cats" topic is a valid topic but has no corresponding articles so you can use this for a 200 test 👍